internal/ingest/ingest.go

```diff
@@ -161,7 +161,7 @@ func (ing *Ingester) Sync(ctx context.Context, peerID peer.ID, peerAddr multiadd
 func (ing *Ingester) watchSyncFinished(onSyncFin <-chan legs.SyncFinished) {
 	for syncFin := range onSyncFin {
 		// Persist the latest sync
-		err := ing.ds.Put(datastore.NewKey(syncPrefix+syncFin.PeerID.String()), syncFin.Cid.Bytes())
+		err := ing.ds.Put(context.Background(), datastore.NewKey(syncPrefix+syncFin.PeerID.String()), syncFin.Cid.Bytes())
```

Comment:
do we want a timeout on these operations?

Reply:
Not for this one.  There is nothing waiting for this to complete, so there is no reason to consider a slow write as too long and need to cancel it.